lib/full-site-editing/block-templates.php

```diff
@@ -195,41 +195,42 @@ function _gutenberg_flatten_blocks( &$blocks ) {
 	return $all_blocks;
 }
 
-/**
- * Parses wp_template content and injects the current theme's
- * stylesheet as a theme attribute into each wp_template_part
- *
- * @param string $template_content serialized wp_template content.
- *
- * @return string Updated wp_template content.
- */
-function _gutenberg_inject_theme_attribute_in_content( $template_content ) {
-	$has_updated_content = false;
-	$new_content         = '';
-	$template_blocks     = parse_blocks( $template_content );
-
-	$blocks = _gutenberg_flatten_blocks( $template_blocks );
-	foreach ( $blocks as &$block ) {
-		if (
-			'core/template-part' === $block['blockName'] &&
-			! isset( $block['attrs']['theme'] )
-		) {
-			$block['attrs']['theme'] = wp_get_theme()->get_stylesheet();
-			$has_updated_content     = true;
+if ( ! function_exists( '_inject_theme_attribute_in_block_template_content' ) ) {
```

Comment:
I think that works until we do some changes in these functions for future releases. It might never be required so I think it's fine for now.

Reply:
In general is it okay to rename function without the GB prefix in the plugin? I guess for functionality that is added through filters or actions we don't rename and bail early like [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/block-patterns.php#L209), but in util functions?

We already have the problem of not being able to change some functions like [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-5.8/index.php#L17).

Are there any instructions for how these cases are handled? --cc @gziolo

Reply:
There are no rules for updating the functionality that is in WP core. If the utility function doesn’t fit anymore you can always deprecate it and create a new one.